.github/workflows/client-compatability-tests.yml

```diff
@@ -0,0 +1,411 @@
+            file: ocr2
+            client: besu            
+            pyroscope_env: ci-smoke-ocr2-evm-simulated      
+          - name: ocr2-erigon
+            run: -run TestOCRv2Basic
+            file: ocr2
+            client: erigon            
+            pyroscope_env: ci-smoke-ocr2-evm-simulated                           
+    runs-on: ubuntu-latest
+    name: Client Compatability Test ${{ matrix.product.name }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.event.merge_group.head_sha }}
+      - name: Build Go Test Command
+        id: build-go-test-command
+        run: |
+          # if the matrix.product.run is set, use it for a different command
+          if [ "${{ matrix.product.run }}" != "" ]; then
+            echo "run_command=${{ matrix.product.run }} ./smoke/${{ matrix.product.file }}_test.go" >> "$GITHUB_OUTPUT"
+          else
+            echo "run_command=./smoke/${{ matrix.product.name }}_test.go" >> "$GITHUB_OUTPUT"
+          fi
+      - name: Check for "enable tracing" label
+        id: check-label
+        run: |
+          label=$(jq -r '.pull_request.labels[]?.name // empty' "$GITHUB_EVENT_PATH")
+
+          if [[ -n "$label" ]]; then
+            if [[ "$label" == "enable tracing" ]]; then
+              echo "Enable tracing label found."
+              echo "trace=true" >> $GITHUB_OUTPUT
+            else
+              echo "Enable tracing label not found."
+              echo "trace=false" >> $GITHUB_OUTPUT
+            fi
+          else
+            echo "No labels present or labels are null."
+            echo "trace=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Setup Grafana and OpenTelemetry
+        id: docker-setup
+        if: steps.check-label.outputs.trace == 'true' && matrix.product.name == 'ocr2' && matrix.product.tag_suffix == '-plugins'
+        run: |
+          # Create network
+          docker network create --driver bridge tracing
+
+          # Make trace directory
+          cd integration-tests/smoke/
+          mkdir ./traces
+          chmod -R 777 ./traces
+
+          # Switch directory
+          cd ../../.github/tracing
+
+          # Create a Docker volume for traces
+          # docker volume create otel-traces
+
+          # Start OpenTelemetry Collector
+          # Note the user must be set to the same user as the runner for the trace data to be accessible
+          docker run -d --network=tracing --name=otel-collector \
+            -v $PWD/otel-collector-ci.yaml:/etc/otel-collector.yaml \
+            -v $PWD/../../integration-tests/smoke/traces:/tracing \
+            --user "$(id -u):$(id -g)" \
+            -p 4317:4317 otel/opentelemetry-collector:0.88.0 --config=/etc/otel-collector.yaml
+      - name: Locate Docker Volume
+        id: locate-volume
+        if: false
```

Comment:
`if: false`? means it never runs? so why keep it?

Reply:
See above